companion/src/simulation/telemetryprovidercrossfire.cpp

```diff
@@ -0,0 +1,530 @@
+
+  QString idFileNameAndPath = QFileDialog::getSaveFileName(this, tr("Save Telemetry"), fldr % "/telemetry.tlm", tr(".tlm Files (*.tlm)"));
+  if (idFileNameAndPath.isEmpty())
+    return;
+
+  QFile file(idFileNameAndPath);
+  if (!file.open(QIODevice::WriteOnly)){
+    QMessageBox::critical(this, CPN_STR_APP_NAME, tr("Unable to open file for writing.\n%1").arg(file.errorString()));
+    return;
+  }
+  QTextStream out(&file);
+
+  out << ui->input_1rss->text();
+  out << "\r\n";
+  out << ui->input_2rss->text();
+  out << "\r\n";
+  out << ui->input_rqly->text();
+  out << "\r\n";
+  out << ui->input_rsnr->text();
+  out << "\r\n";
+  out << ui->input_trss->text();
+  out << "\r\n";
+  out << ui->input_tpwr->currentText();
+  out << "\r\n";
+  out << ui->input_rfmd->text();
+  out << "\r\n";
+  out << ui->input_ant->text();
+  out << "\r\n";
+  out << ui->input_tqly->text();
+  out << "\r\n";
+  out << ui->input_tsnr->text();
+  out << "\r\n";
+  out << ui->input_rrsp->text();
+  out << "\r\n";
+  out << ui->input_rpwr->text();
+  out << "\r\n";
+  out << ui->input_trsp->text();
+  out << "\r\n";
+  out << ui->enabled_battery->isChecked();
+  out << "\r\n";
+  out << ui->input_rxbt->text();
+  out << "\r\n";
+  out << ui->input_curr->text();
+  out << "\r\n";
+  out << ui->input_capa->text();
+  out << "\r\n";
+  out << ui->input_batpercent->text();
+  out << "\r\n";
+  out << ui->enabled_gps->isChecked();
+  out << "\r\n";
+  out << ui->input_gps->text();
+  out << "\r\n";
+  out << ui->input_gspd->text();
+  out << "\r\n";
+  out << ui->input_hdg->text();
+  out << "\r\n";
+  out << ui->input_sats->text();
+  out << "\r\n";
+  out << ui->enabled_attitude->isChecked();
+  out << "\r\n";
+  out << ui->input_ptch->text();
+  out << "\r\n";
+  out << ui->input_roll->text();
+  out << "\r\n";
+  out << ui->input_yaw->text();
+  out << "\r\n";
+  out << ui->enabled_flightcontroller->isChecked();
+  out << "\r\n";
+  out << ui->input_fm->text();
+  out << "\r\n";
+  out << ui->enabled_barometer->isChecked();
+  out << "\r\n";
+  out << ui->input_vspd->text();
+  out << "\r\n";
+  out << ui->input_alt->text();
+  out << "\r\n";
+  file.flush();
+  file.close();
+}
+
+void TelemetryProviderCrossfire::on_button_loadTelemetryValues_clicked()
+{
+  QString fldr = g.backupDir().trimmed();
+  if (fldr.isEmpty())
+    fldr = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+
+  QString idFileNameAndPath = QFileDialog::getOpenFileName(this, tr("Open Telemetry File"), fldr % "/telemetry.tlm", tr(".tlm Files (*.tlm)"));
+  if (idFileNameAndPath.isEmpty())
+    return;
+
+  QFile file(idFileNameAndPath);
+
+  if (!file.open(QIODevice::ReadOnly)){
+    QMessageBox::critical(this, CPN_STR_APP_NAME, tr("Unable to open file for reading.\n%1").arg(file.errorString()));
+    return;
+  }
+
+  QTextStream in(&file);
+
+  QString inputText;
+  double inputDouble;
+  int inputInt;
+
```

Comment:
Currently possible to inadvertently load Frsky saved values. Needs a test and rejection msg to save the user from themselves